datafusion/execution/src/object_store.rs

```diff
@@ -156,78 +158,64 @@ impl std::fmt::Debug for ObjectStoreRegistry {
     }
 }
 
-impl Default for ObjectStoreRegistry {
+impl Default for DefaultObjectStoreRegistry {
     fn default() -> Self {
         Self::new()
     }
 }
 
-impl ObjectStoreRegistry {
-    /// Create an [`ObjectStoreRegistry`] with no [`ObjectStoreProvider`].
-    ///
-    /// This will register [`LocalFileSystem`] to handle `file://` paths, further stores
-    /// will need to be explicitly registered with calls to [`ObjectStoreRegistry::register_store`]
+impl DefaultObjectStoreRegistry {
+    /// This will register [`LocalFileSystem`] to handle `file://` paths
     pub fn new() -> Self {
-        ObjectStoreRegistry::new_with_provider(None)
-    }
-
-    /// Create an [`ObjectStoreRegistry`] with the provided [`ObjectStoreProvider`]
-    ///
-    /// This will register [`LocalFileSystem`] to handle `file://` paths, further stores
-    /// may be explicity registered with calls to [`ObjectStoreRegistry::register_store`] or
-    /// created lazily, on-demand by the provided [`ObjectStoreProvider`]
-    pub fn new_with_provider(provider: Option<Arc<dyn ObjectStoreProvider>>) -> Self {
         let object_stores: DashMap<String, Arc<dyn ObjectStore>> = DashMap::new();
-        object_stores.insert("file://".to_string(), Arc::new(LocalFileSystem::new()));
-        Self {
-            object_stores,
-            provider,
-        }
+        let url = Url::parse("file://").unwrap();
+        let key = get_url_key(&url);
+        object_stores.insert(key, Arc::new(LocalFileSystem::new()));
```

Comment:
```suggestion
        object_stores.insert("file://".to_string(), Arc::new(LocalFileSystem::new()));
```

See below comment about removing trailing `/`